instrumentation/methods/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/methods/MethodSpanNameExtractor.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.methods;
+
+import static io.opentelemetry.instrumentation.api.tracer.BaseTracer.spanNameForMethod;
```

Comment:
Instead of using the method from `BaseTracer`, let's add to `instrumentation` API when needed. I think generating a span name from a method is common so we could have a util class / unit tests for it.

Reply:
I'm a little confused why is `BaseTracer.spanNameForMethod` method declared `public static`. How about moving these common methods(eg. spanNameForMethod) to ` io.opentelemetry.javaagent.tooling.Utils `. WDYT? @anuraaga : )

Reply:
It being public static is legacy, it should probably have a module but we just added it to BaseTracer organically.

We don't want any instrumentation utils in `javaagent-tooling` since they're used outside the agent. It would go in `instrumentation-api`. How about adding a `SpanNames` class there fo rnow

Reply:
By the way for some context, while ideally we just have a `SpanNameExtractor<Method>` in that module itself, I'm not sure we'd be able to handle cases like this one without a util class anyways

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/jaxrs/jaxrs-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v1_0/JaxRsAnnotationsTracer.java#L50

But maybe others have some thoughts on that too

Reply:
> By the way for some context, while ideally we just have a `SpanNameExtractor<Method>` in that module itself, I'm not sure we'd be able to handle cases like this one without a util class anyways

We'll have to think of something once we get there 😅 
Ideally I'd like us to have `SpanNameExtractor`s for class/method named spans (and `AttributeExtractor` for `code.namespace` & `code.function`) in instrumentation-api, but we can think of how to implement them in one of the next PRs.